### PR TITLE
Spring App

### PR DESCRIPTION
**Pull request by VishovVladimir**
What made: 
Spring MVC framework app.
Two endpoints.
Test these.
Generation swagger UI and OpenAPI.
build to docker image

### CODE REVIEW

**Review by shibanovp — COMMENTED** [review]
@VishovVladimir, this looks great! I've left a few minor, non-blocking comments.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        run: ./gradlew clean build
+
+      - name: Docker build image
+        run: docker build -t java-starter-template:latest .
+
+      - name: Tag Docker image
+        run: docker tag java-starter-template:latest ghcr.io/experienced-dev/java-starter-template/builds:${{ github.sha }}
+
+      - name: Push Docker image to Docker Hub
+        run: docker push ghcr.io/experienced-dev/java-starter-template/builds:${{ github.sha }}
```
[the remark this answers is not in the document]
Comment (VishovVladimir) [reply .github/workflows/build.yaml:44]: What you mean?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM openjdk:17-jdk-slim
+
+COPY build/libs/*.jar /app.jar
+
+# Set JVM memory restriction to 1GB
+ENTRYPOINT ["java", "-Xmx1024m", "-jar", "/app.jar"]
```
[the remark this answers is not in the document]
Comment (VishovVladimir) [reply Dockerfile:6]: Yes it does. I will use JVM_MAX_MEMORY

File: src/test/java/experienced/dev/javastartertemplate/TemplateControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        mockMvc.perform(get("/healthz"))
+                .andExpectAll(status().isOk()); // HTTP_OK = 200
+    }
+
+    @Test
+    void readyCheck() throws Exception {
+        mockMvc.perform(get("/readyz"))
+                .andExpectAll(status().isOk()); // HTTP_OK = 200
+    }
+}
```
[the remark this answers is not in the document]
Comment (VishovVladimir) [reply src/test/java/experienced/dev/javastartertemplate/TemplateControllerTest.java:27]: What u mean? Move comment to next line?

File: .github/workflows/build.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (shibanovp) [reply .github/workflows/build.yaml:44]: https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

GitHub indicates that with a red icon.
image

File: HELP.md
[the remark this answers is not in the document]
Comment (VishovVladimir) [reply HELP.md:1]: Ok. I will do

File: src/test/java/experienced/dev/javastartertemplate/TemplateControllerTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (shibanovp) [reply src/test/java/experienced/dev/javastartertemplate/TemplateControllerTest.java:27]: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (shibanovp) [reply Dockerfile:6]: I think we could just set it on a Dockerfile level with ENV

**Review by shibanovp — APPROVED** [review]

**Pull request merged**